src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs

```diff
@@ -212,18 +220,18 @@ private NetworkPeerConnectionParameters CreateNetworkPeerConnectionParameters()
         /// Check if the client is allowed to connect based on certain criteria.
         /// </summary>
         /// <returns>When criteria is met returns <c>true</c>, to allow connection.</returns>
-        private bool AllowClientConnection(TcpClient tcpClient)
+        private (bool successful, string reason) AllowClientConnection(TcpClient tcpClient)
         {
             if (this.networkPeerDisposer.ConnectedInboundPeersCount >= this.connectionManagerSettings.MaxInboundConnections)
             {
                 this.logger.LogTrace("(-)[MAX_CONNECTION_THRESHOLD_REACHED]:false");
-                return false;
+                return (false, "Max Connection Threshold Reached.");
```

Comment:
Lets make these messages more structured. Change the string to: `Inbound Refused: Max Connection Threshold Reached.`

Reply:
done